src/parser.y

```diff
@@ -345,6 +349,26 @@ FuncDef Exp %prec FUNCDEF {
   $$ = block_bind_referenced($1, $2, OP_IS_CALL_PSEUDO);
 } |
 
+"codef" '(' '$' ')' IDENT ':' Exp ';' %prec FUNCDEF Exp {
```

Comment:
Why do we need to add the `($)` prefix? I suspect that's due to the fact that two symbols are bound, both `IDENT` and `$IDENT`. In any case, it looks awkward IMO

Reply:
If you would like to have the prefix then maybe we should still use the standard `def` keyword. That would make more sense to me then.
```
def test: .+1;  // function
def ($)test: .+1; // coroutine
```

Reply:
I agree it looks awkward.  I don't recall why I did it that way.  Maybe it was to not force `codef` to be a new strong keyword?

Reply:
And yes, you're right, I could drop the `co` from `codef` given this weird `($)` thing.

Reply:
Oh and I had wanted to indicate lexically somehow that this creates bindings for `name` and `$name`, just like `def f($name): ...;` does in the body of `f`.  The latter apparently is not always obvious, and there's an open issue about that that I think I'll just close as will-not-fix because it's too late to do anything about it (there the user was surprised by `def f($length): ...;` shadowing the builtin `length`!).

Reply:
Yeah, that's what I've suspected. But I don't think that this actually makes it clear to the user without reading documentation, and as such, it's just an unnecessary complication of code entry.

On the other hand, there is a completely different syntax for converting parameter expressions to coexpressions with the help of `@@`. 
I think that maybe it would be useful to syntactically relate the two features to help the user connect them

Reply:
For example, having `@` in both cases:
```
def @coroutine: . + 1;
def func(@coarg): . + 1;
```

Reply:
Sold.  I'll go with that!

Reply:
we can even go further and have `@(expr)` instead of `coexp(expr)` to close the loop